pilot/pkg/model/authentication.go

```diff
@@ -213,3 +174,65 @@ func constructgRPCCallCredentials(tokenFileName, headerKey string) []*core.GrpcS
 		},
 	}
 }
+
+func constructSdsSecretConfigHelper(name string, grpcConfig *core.GrpcService_GoogleGrpc) *auth.SdsSecretConfig {
+	if name == "" || grpcConfig == nil {
+		return nil
+	}
+	return &auth.SdsSecretConfig{
+		Name: name,
+		SdsConfig: &core.ConfigSource{
+			ConfigSourceSpecifier: &core.ConfigSource_ApiConfigSource{
+				ApiConfigSource: &core.ApiConfigSource{
+					ApiType: core.ApiConfigSource_GRPC,
+					GrpcServices: []*core.GrpcService{
+						{
+							TargetSpecifier: &core.GrpcService_GoogleGrpc_{
+								GoogleGrpc: grpcConfig,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func constructGrpcConfig(sdsUdsPath string, addCredential bool, useK8sSATrustworthyJwt, useK8sSANormalJwt bool) *core.GrpcService_GoogleGrpc {
+	if sdsUdsPath == "" {
+		return nil
+	}
+
+	gRPCConfig := &core.GrpcService_GoogleGrpc{
+		TargetUri:  sdsUdsPath,
+		StatPrefix: SDSStatPrefix,
+	}
+
```

Comment:
remove this

Reply:
This is for ingress gateway to speak gRPC to gateway agent without credentials.

Reply:
I see, better to have a separated func for ingress case, will be much simpler

Reply:
Done. Thanks.